pkg/vulnerabilityreport/builder.go

```diff
@@ -68,11 +70,21 @@ func (s *ScanJobBuilder) WithCredentials(credentials map[string]docker.Auth) *Sc
 	return s
 }
 
+func (s *ScanJobBuilder) WithClient(client client.Client) *ScanJobBuilder {
+	s.client = client
+	return s
+}
+
 func (s *ScanJobBuilder) Get() (*batchv1.Job, []*corev1.Secret, error) {
 	spec, err := kube.GetPodSpec(s.object)
 	if err != nil {
 		return nil, nil, err
 	}
+	podSpecHash := kube.ComputeHash(spec)
+
+	// get nodename from running pods.
+	resolver := kube.ObjectResolver{Client: s.client}
+	spec.NodeName, err = resolver.GetNodeName(context.Background(), s.object)
```

Comment:
Shouldn't we do that only if trivy fs is enabled? I'm not sure that the job builder is the right place to resolve node name anyway. I'd do that within Trivy plugin, which returns scan job spec with node name (if required by config).